### PR TITLE
AMBARI-23456. Add developement guide markdowns & remove some old docs

### PR DESCRIPTION
**Pull request by oleewere**
# What changes were proposed in this pull request?
- add development guide (backend only)
- remove logsearch/logfeeder readme.md (move development guides to development.md)
- edit README.md to have similar content that docs/index.md has

## How was this patch tested?
mkdocs serve

Please review @g-boros

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-LogSearch-Github-PR-Builder/128/
Test PASSed.

File: README.md [hunk 1]
```diff
@@ -2,89 +2,38 @@
...
-export LOGSEARCH_JDK_11=true
-make deb
-```
-3. Generated RPM/DPKG files will be found in ambari-logsearch-assembly/target folder
+## Features
 
-## Running Integration Tests
+- Parse / aggregate and ship logs
+- Send and index logs in Solr
+- Store logs (structured or un-sructured format) in Cloud Storage (S3 / GCS / ADLS / WASB)
```
Comment (adoroszlai) [inline README.md]: `un-sructured` -> `unstructured`

File: README.md [hunk 2]
```diff
@@ -2,89 +2,38 @@
...
+- Fultext Search in logs (if the logs are shipped to Solr)
```
Comment (adoroszlai) [inline README.md]: `Fultext` -> `Full-text`

File: README.md [hunk 3]
```diff
@@ -2,89 +2,38 @@
...
-# from ambari-logsearch folder
-mvn clean integration-test -Dbackend-tests failsafe:verify
-# or run selenium tests with docker for mac, but before that you nedd to start xquartz
-open -a XQuartz
-# then in an another window you can start ui tests
-mvn clean integration-test -Dselenium-tests failsafe:verify
-# you can specify story file folde location with -Dbackend.stories.location and -Dui.stories.location (absolute file path) in the commands
-```
-Also you can run from the IDE, but make sure all of the ambari logsearch modules are built.
+- Log Feeder: agent component on all hosts to monitor and shipping logs.
```
Comment (adoroszlai) [inline README.md]: `monitor and shipping` should have the same form.

File: README.md [hunk 4]
```diff
@@ -2,89 +2,38 @@
...
 
+![Log Search Architecture Overview](docs/images/architecture_overview.jpg)
 
-## Update version (for release or specific builds)
+## Contents
 
-```bash
-make update-version new-version="2.8.0.0-11"
-```
+- [1, Installation](docs/installation.md)
```
Comment (adoroszlai) [inline README.md]: `1,` -> `1.`

File: docs/development.md [hunk 5]
```diff
@@ -15,9 +15,124 @@ See the License for the specific language governing permissions and
 limitations under the License.
 -->
 
-### Backend development
+## Developement guide
```
Comment (adoroszlai) [inline docs/development.md]: `Developement` -> `Development`

File: docs/development.md [hunk 6]
```diff
@@ -15,9 +15,124 @@ See the License for the specific language governing permissions and
...
+
+- Install [docker](https://docs.docker.com/)
+- For Mac OS X use [Docker for Mac](https://docs.docker.com/docker-for-mac/)
+- [Docker compose](https://docs.docker.com/compose/) is also required.
+
+### Build and start Log Search in docker container
+```bash
+# to see available commands: run start-logsearch without arguments
+cd docker
+./logsearch-docker build-and-run # build mvn project locally, build docker image, start containers
```
Comment (adoroszlai) [inline docs/development.md]: Command is `logsearch-docker`, but comment above says `start-logsearch`.  Shouldn't they match?

File: docs/development.md [hunk 7]
```diff
@@ -15,9 +15,124 @@ See the License for the specific language governing permissions and
...
+You can also start Log Search server from an IDE as well. One thing is important: the config set location that the server tries to upload to ZooKeeper. By default config sets are located at `${LOGSEARCH_SERVER_RELATIVE_LOCATION:}src/main/configsets` in `logsearch.properties`. Based or from where you run `LogSearch.java`, you need to set `LOGSEARCH_SERVER_RELATIVE_LOCATION` env variable properly. (or just simply use the ambari-logsearch-server as the working directory)
+
+#### Start Log Feeder locally from maven / IDE
+
+First you need to start every required service (except logfeeder), go to `ambari-logsearch/docker` folder and run:
+```bash
+docker-compose up -d zookeeper solr logsearch
+```
+
+Secondly, if you are planning to run Log Feeder from an IDE, for running the LogFeeder main methoud, you will need to set the working directory to `ambari/ambari-logsearch/ambari-logsearch-logfeeder` or set `LOGFEEDER_RELATIVE_LOCATION` env variable.
```
Comment (adoroszlai) [inline docs/development.md]: `methoud` -> `method`

File: docs/development.md [hunk 8]
```diff
@@ -15,9 +15,124 @@ See the License for the specific language governing permissions and
...
+make deb
+# or for jdk11
+export LOGSEARCH_JDK_11=true
+make deb
+```
+3. Generated RPM/DPKG files will be found in ambari-logsearch-assembly/target folder
+
+### Running Integration Tests
+
+By default integration tests are not a part of the build process, you need to set -Dbackend-tests or -Dselenium-tests (or you can use -Dall-tests to run both). To running the tests you will need docker here as well (right now docker-for-mac and unix are supported by default, for boot2docker you need to pass -Ddocker.host parameter to the build).
```
Comment (adoroszlai) [inline docs/development.md]: `To running` -> `To run`

File: docs/development.md [hunk 9]
```diff
@@ -15,9 +15,124 @@ See the License for the specific language governing permissions and
...
+
+```bash
+# from ambari-logsearch folder
+mvn clean integration-test -Dbackend-tests failsafe:verify
+# or run selenium tests with docker for mac, but before that you nedd to start xquartz
+open -a XQuartz
+# then in an another window you can start ui tests
+mvn clean integration-test -Dselenium-tests failsafe:verify
+# you can specify story file folde location with -Dbackend.stories.location and -Dui.stories.location (absolute file path) in the commands
```
Comment (adoroszlai) [inline docs/development.md]: `story file folde location` seems to be redundant, `story file location` or `story file folder` would be enough.

**Pull request merged**